prov/gni/src/gnix_mr_cache.c

```diff
@@ -508,11 +796,18 @@ static inline int __mr_cache_entry_put(
 		gnix_mr_cache_entry_t *entry)
 {
 	RbtIterator iter;
+	int rc;
 	gni_return_t grc = GNI_RC_SUCCESS;
 	RbtIterator found;
 	gnix_mr_cache_entry_t *parent = NULL;
 	struct dlist_entry *next;
 
+	GNIX_TRACE(FI_LOG_MR, "\n");
+
+	if (cache->attr.lazy_deregistration) {
+		__clear_notifier_events(cache);
+	}
```

Comment:
Is this going to handled in a later PR? I can't quite recall. 

This is related to the discussion that notifier events may pertain to registrations in a non-lazy cache as well.